src/framework-configuration.js

```diff
@@ -256,17 +303,34 @@ export class FrameworkConfiguration {
   /**
    * Configures an external, 3rd party plugin before Aurelia starts.
    * @param plugin The ID of the 3rd party plugin to configure.
-   * @param config The configuration for the specified plugin.
+   * @param pluginConfig The configuration for the specified plugin.
    * @return Returns the current FrameworkConfiguration instance.
  */
-  plugin(plugin: string, config?: any): FrameworkConfiguration {
+  plugin(
+    plugin: string | ((frameworkConfig: FrameworkConfiguration) => any) | FrameworkPluginInfo,
+    pluginConfig?: any
+  ): FrameworkConfiguration {
     assertProcessed(this);
 
-    if (typeof (plugin) === 'string') {
-      return this.plugin({ moduleId: plugin, resourcesRelativeTo: [plugin, ''], config: config || {} });
+    let info: FrameworkPluginInfo;
+    switch (typeof plugin) {
+    case 'string':
+      info = { moduleId: plugin, resourcesRelativeTo: [plugin, ''], config: pluginConfig || {} };
+      break;
+    case 'function':
+      if (this.queuedPlugins.has(plugin)) {
```

Comment:
@EisenbergEffect Not sure if we should use array instead of WeakSet here. Pros: IE11 can run without collection polyfill (doubt anyone do that though 😄 )

Reply:
If we can avoid WeakSet, that would be best. While we ship the polyfills, I don't think we use any of the weak collections anywhere.

Reply:
@EisenbergEffect switched to array